### ISSUE TITLE
Override for Gradient Clipping

### ISSUE THREAD

**Issue by ericharper** [PyTorchLightning/pytorch-lightning#3912]
## 🚀 Feature
We would like to be able to implement our own gradient clipping using a `LightningModule` override similar to `configure_sync_batchnorm` or `configure_ddp`, see [advanced-hooks](https://pytorch-lightning.readthedocs.io/en/stable/lightning-module.html#advanced-hooks). 

### Motivation

We are implementing Model Parallel [Megatron-LM](https://github.com/NVIDIA/Megatron-LM) models as LightningModules for [NeMo](https://github.com/NVIDIA/NeMo). Gradient clipping is very important for training extremely large transformer models and when using model parallelism, gradient's need to be clipped across model parallel groups. This means we need to be able to `configure_gradient_clipping` after `torch.distributed` has been initialized (so we have access to the model parallel groups). Here is the Megatron-LM gradient clipping implementation for [reference](https://github.com/NVIDIA/Megatron-LM/blob/19301985dd31c8b612095cbad15bd903e8ddd497/megatron/mpu/grads.py#L74)

We've already taken advantage of PTL's advanced hooks as you can see in our in-progress implementation [here](https://github.com/NVIDIA/NeMo/blob/e44d5e8f2b8d6ffa6a02fd0510e5b004d66e772a/nemo/collections/nlp/models/nlp_model.py#L41) and [here](https://github.com/NVIDIA/NeMo/blob/e44d5e8f2b8d6ffa6a02fd0510e5b004d66e772a/nemo/collections/nlp/models/nlp_model.py#L66).

### Pitch

Add `configure_gradient_clipping` hook.

Be able to write custom implementation for: https://github.com/PyTorchLightning/pytorch-lightning/blob/c510a7f90077140d60c47adf8e1e73638c2d1017/pytorch_lightning/accelerators/base_backend.py#L140

Important that it's called after ddp is initialized: https://github.com/PyTorchLightning/pytorch-lightning/blob/c510a7f90077140d60c47adf8e1e73638c2d1017/pytorch_lightning/accelerators/ddp_backend.py#L231

### Additional context

**Comment by ydcjeff** [comment]
@PyTorchLightning/core-contributors sounds good?

**Comment by rohitgr7** [comment]
IMO it should be a part of LightningModule since gradients are involved here.

**Comment by carmocca** [comment]
Duplicate of https://github.com/PyTorchLightning/pytorch-lightning/issues/5982

Closing in favor of that one since it has more discussion
